source/extensions/http/header_validators/envoy_default/config_overrides.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "envoy/server/factory_context.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Http {
+namespace HeaderValidators {
+namespace EnvoyDefault {
+
+class ConfigOverrides {
+public:
+  ConfigOverrides() = default;
+  ConfigOverrides(Server::Configuration::ServerFactoryContext& server_context)
+      : preserve_url_encoded_case_(server_context.runtime().snapshot().getBoolean(
+            "envoy.uhv.preserve_url_encoded_case", true)) {}
+
+  // This flag enables preservation of the case of percent-encoded triplets in URL path for
+  // compatibility with legacy path normalization.
+  // https://datatracker.ietf.org/doc/html/rfc3986#section-2.1 mandates that uppercase
+  // hexadecimal digits (A through F) are equivalent to lowercase.
+  // However to make path matching of percent-encoded triplets easier path normalization changes all
+  // hexadecimal digits to uppercase.
+  //
+  // This option currently is `true` by default and can be overridden using the
+  // "envoy.uhv.preserve_url_encoded_case" runtime value. Note that the default value
+  // will be changed to `false` in the future to make it easier to write path matchers that
+  // look for percent-encoded triplets.
+  bool preserve_url_encoded_case_{true};
```

Comment:
nit: If this is a class, then the members should be private. Also, could this be const?

Reply:
Changed it to `struct` since per style guide types without behavior can use `struct`. Less boiler plate.

Made the member `const`.